modules/basegl/include/modules/basegl/shadercomponents/tfcomponent.h

```diff
@@ -0,0 +1,73 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2022 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <modules/basegl/baseglmoduledefine.h>
+
+#include <inviwo/core/ports/volumeport.h>                     // for VolumeInport
+#include <inviwo/core/properties/invalidationlevel.h>         // for InvalidationLevel, Invalida...
+#include <inviwo/core/properties/transferfunctionproperty.h>  // for TransferFunctionProperty
+#include <modules/basegl/shadercomponents/shadercomponent.h>  // for ShaderComponent::Segment
+#include <modules/opengl/shader/shader.h>                     // for Shader
+#include <modules/opengl/shader/shaderobject.h>               // for ShaderObject
+#include <modules/opengl/texture/textureunit.h>               // for TextureUnit, TextureUnitCon...
+#include <modules/opengl/image/layergl.h>                     // IWYU pragma: keep
+#include <modules/opengl/texture/textureutils.h>
+
+#include <algorithm>  // for max
+
+#include <cstddef>      // for size_t
+#include <string>       // for string
+#include <string_view>  // for string_view
+#include <vector>       // for vector
+
+#include <fmt/core.h>  // for format
+
+namespace inviwo {
+
+/**
+ * Adds a TransferFunctionProperty, and binds it to uniforms in the shader.
+ */
+class TFComponent : public ShaderComponent {
```

Comment:
Would also make this consistent with other naming, e.g., TransferFunctionProperty 
```suggestion
class TransferFunctionComponent : public ShaderComponent {
```

Reply:
The reason for `TransferFunction` is legacy since we did not want to break too many things when introducing the IsoTFProperty.